packages/webui/src/client/ui/UserEditOperations/PropertiesPanel.tsx

```diff
@@ -0,0 +1,549 @@
+											)
+										case UserEditingType.FORM:
+											return (
+												<EditingTypeChangeForm
+													key={i}
+													userEditOperation={userEditOperation}
+													segment={segment}
+													part={part}
+													rundownId={rundownId}
+													pendingChanges={pendingChanges}
+													setPendingChanges={setPendingChanges}
+												/>
+											)
+										case UserEditingType.SOURCE_LAYER_FORM:
+											return (
+												<EditingTypeChangeSourceLayerSource
+													key={i}
+													userEditOperation={userEditOperation}
+													segment={segment}
+													part={part}
+													rundownId={rundownId}
+													pendingChanges={pendingChanges}
+													setPendingChanges={setPendingChanges}
+												/>
+											)
+										default:
+											assertNever(userEditOperation)
+											return null
+									}
+								})}
+						</div>
+					</>
+				)}
+				<div className="propertiespanel-pop-up__footer">
+					<button
+						className="propertiespanel-pop-up__button"
+						onClick={handleRevertChanges}
+						disabled={!hasPendingChanges}
+					>
+						<span className="propertiespanel-pop-up__label">REVERT CHANGES</span>
+					</button>
+					<button
+						className="propertiespanel-pop-up__button"
+						onClick={handleCommitChanges}
+						disabled={!hasPendingChanges}
+					>
+						<span className="propertiespanel-pop-up__label">COMMIT CHANGES</span>
+					</button>
+				</div>
+			</div>
+		</div>
+	)
+}
+
+function EditingTypeAction(props: {
+	userEditOperation: CoreUserEditingDefinitionAction
+	segment: DBSegment | undefined
+	part: DBPart | undefined
+	rundownId: RundownId
+	pendingChanges: PendingChange[]
+	setPendingChanges: React.Dispatch<React.SetStateAction<PendingChange[]>>
+}) {
+	const getPendingState = () => {
+		const pendingChange = props.pendingChanges.find((change) => change.operationId === props.userEditOperation.id)
+		return pendingChange?.switchState
+	}
+
+	const [hasBeenEdited, setHasBeenEdited] = React.useState<boolean>(
+		getPendingState() !== undefined && getPendingState() !== props.userEditOperation.isActive
+	)
+
+	React.useEffect(() => {
+		setHasBeenEdited(getPendingState() !== undefined && getPendingState() !== props.userEditOperation.isActive)
+	}, [props.userEditOperation.id, props.pendingChanges])
+
+	if (!props.userEditOperation.buttonType) return null
+	const addPendingChange = () => {
+		setHasBeenEdited(!hasBeenEdited)
+		props.setPendingChanges((prev) => {
+			// Find if there's an existing pending change for this operation
+			const existingChangeIndex = prev.findIndex((change) => change.operationId === props.userEditOperation.id)
+
+			if (existingChangeIndex !== -1) {
+				// If exists, toggle the switch state
+				const newChanges = [...prev]
+				newChanges[existingChangeIndex] = {
+					...newChanges[existingChangeIndex],
+					switchState: !newChanges[existingChangeIndex].switchState,
+				}
```

Comment:
I don't think I agree with this behaviour.
On second click of this, it will be executing the operation to change the state to the current value.

And replacing the earlier change can cause some unintended behaviour.
If the user clicks:
1) toggle theme graphic
2) disable all graphics
3) toggle theme graphic (this should enable the theme graphic)

then that will be executed as
1) toggle theme graphic
2) disable all graphics (this now undoes the toggle that was supposed to follow it)

making changes to the queued operations is only ok if we can guarantee that the operations are not dependent on each other. 
but also I understand why you have done that, as we don't want to build a queue of hundreds of operations because the user was clicking around while figuring out what they are trying to do.